### PR TITLE
Port `bin/delete-version` to use Diesel

### PR DESCRIPTION
**Pull request by sgrif**
The changes here were broadly similar to those in `delete-crate`. The
child row deletions are no longer needed here, as they cascade in the
database. (Arguably neither of these binaries are really needed anymore
since it can just be done in the database)

### CODE REVIEW

File: src/bin/delete-version.rs [hunk 1]
```diff
@@ -8,28 +8,29 @@
...
-    }
+    let conn = cargo_registry::db::connect_now().unwrap();
+    conn.transaction::<_, diesel::result::Error, _>(|| {
+        delete(&conn);
+        Ok(())
+    }).unwrap()
 }
 
-fn delete(tx: &postgres::transaction::Transaction) {
+fn delete(conn: &PgConnection) {
```
Comment (vignesh-sankaran) [inline src/bin/delete-version.rs]: Would it make sense to add doc comments to functions in the `bin/` directory?

**Comment by sgrif** [comment]
@vignesh-sankaran Have you run the migrations? I cannot replicate, and it should be impossible to violate that FK

File: src/bin/delete-version.rs (hunk 1 above)
Comment (carols10cents) [inline src/bin/delete-version.rs]: I think for the scripts in src/bin, adding documentation more like a man page using docopt or clap would be more appropriate and useful than doc comments. WDYT?

**Pull request merged**